LEAF_Request_Portal/js/parallelProcessing.js

```diff
@@ -46,17 +46,19 @@ function parallelProcessing(recordID, orgChartPath, CSRFToken)
         $.ajax({
             type: 'GET',
             url: 'api/?a=form/'+recordID+'/workflow/indicator/assigned',
-            success: function(obj) {
-                indicatorObject = obj;
-                for (var i = 0; i < indicatorObject.length; i++) {
-                    $(document.createElement('option'))
-                        .attr('value', indicatorObject[i].indicatorID)
-                        .html(indicatorObject[i].name)
-                        .appendTo($("select#indicator_selector"));
+            success: function(arr) {
+                indicatorObject = arr;
+                if (indicatorObject !== null) {
+                    for (let i = 0; i < indicatorObject.length; i++) {
+                        $(document.createElement('option'))
+                            .attr('value', indicatorObject[i].indicatorID)
+                            .html(indicatorObject[i].name)
+                            .appendTo($("select#indicator_selector"));
+                    }
                 }
-                if(obj.length == 0) {
+                else {
                     $('#submitControl').css('display', 'none');
-                    $('#selectDiv').html('<span style="font-size: 120%">Error: The form must contain a field of type "orgchart group" or "orgchart employee" to begin Parallel Processing.</span>');
+                    $('#selectDiv').html('<span style="font-size: 120%">Error: The form/workflow must contain a field of type "orgchart group" or "orgchart employee" to begin Parallel Processing.</span>');
```

Comment:
Let's start pulling text out of the line that uses them.  Eventually they are going to be supplied from a completely different resource based on using a response template.  So in this case, inside the "else" clause create a "failResponse" variable and assign the response text to it.  And then an "htmlResponse" variable that has the rest of what is currently in the .html() and uses the failResponse variable.  Then with the $('#selectDiv').html(htmlResponse).  I know that seems like a lot of work for something that is simple, but it's going to start to pay off big time when we go to move things out.

Reply:
👍 Changed